### PR TITLE
Extract obsolete operating systems from LDAP

### PR DESCRIPTION
**Pull request by Shad0wC0ntr0ller**
Extract enabled obsolete operating systems from LDAP

nxc ldap dcip -u user -p password -M obsolete

image

### CODE REVIEW

**Comment by zblurx** [comment]
Hey, I love this idea
However for parsing purpose, choosing arrow as the separator is not great (how can you awk the output ?)
What about `machinename (10.10.10.1) : OS name` ?
Or maybe a module option to set a delimiter or even a csv output file ?

**Comment by Shad0wC0ntr0ller** [comment]
@zblurx you know, that’s a great point that has not been brought to my attention before. I will update the module in an hour or so. Thanks for the good call out.

**Comment by Shad0wC0ntr0ller** [comment]
@zblurx I have made the adjustment. I also went ahead and added Server 2012 as that's less than a couple of weeks away. Finally i updated the output path from cme to nxc.

**Comment by NeffIsBack** [comment]
Nice! Also it would probably be nice to give some feedback to the user about the file you are saving the output to. Something like sam and lsa dump do

**Comment by Shad0wC0ntr0ller** [comment]
There is an info statement that shows where the file is saved to in the output already. 

image

Are you saying you want them to get added to a database like this ?

image

**Comment by NeffIsBack** [comment]
Oh my bad, missed that line. Just wanted to have the file path in some output but you already thought of that😄

File: nxc/modules/obsolete.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            except Exception as e:
+                context.log.debug("Exception encountered:", exc_info=True)
+                context.log.debug(f'Skipping item, cannot process due to error {str(e)}')
+
+        if answers:
+            
+            hostname_parts = answers[0][0].split('.')
+            domain = ".".join(hostname_parts[1:])
+            
+            home = Path.home()
```
Comment (zblurx) [inline nxc/modules/obsolete.py:68]: There is a more conventional way to get the logs folders in NXC -> `log_folder_path = os.path.join(os.path.expanduser("~/.nxc"), "logs")`

**Comment by evildrummer** [comment]
Maybe adding the pwdlastset of the host? So you know it if the host is "kind a active"

**Comment by Shad0wC0ntr0ller** [comment]
@evildrummer @zblurx Excellent ideas, I will look into this later today !

**Comment by NeffIsBack** [comment]
@Shad0wC0ntr0ller any update on this?

**Comment by Shad0wC0ntr0ller** [comment]
Sorry guys been busy with other things, ill get this handled by EOD today

**Comment by NeffIsBack** [comment]
> Sorry guys been busy with other things, ill get this handled by EOD today

Awesome, but no pressure, justed wanted to check :)

**Comment by Marshall-Hallenbeck** [comment]
@Shad0wC0ntr0ller did you want to get this up to date and working? We'd like to include it in the v1.2 release if you have time :)

**Comment by Shad0wC0ntr0ller** [comment]
I will try to make the needed updates over the next couple of days. Thank you

**Comment by Shad0wC0ntr0ller** [comment]
Sorry for the delay, I have updated the script it will now include the pwdlastset variable. let me know if there's any more adjustments you would like to see. thank you

**Review by NeffIsBack — APPROVED** [review]
Very nice addition! Thanks again for you work

**Comment by NeffIsBack** [comment]
Working flawlessly:
image

**Comment by enj5oy** [comment]
@Shad0wC0ntr0ller hello man, can you explain when this info can help?

**Comment by evildrummer** [comment]
@enj5oy it was a request from me. In PenTests you always check fir legacy or unsupported systems for the client.
It could also be a quick overview for easy targets

**Pull request merged**